### PR TITLE
Degree days remix

### PR DESCRIPTION
**Pull request by charparr**
This PR reflects a change in the underlying source dataset and coverage for the degree day endpoints including freezing index, thawing index, heating degree days, and degree days below zero. More models and scenarios are now available.

Notes
 - Please point your Rasdaman URL to Zeus for testing.
 - I did add a function to the `postprocessing` module to strictly eliminate keys in the response with no data.
 - Please review the template documentation - I had to make some changes based on the new source dataset.

What to test:

Well, the thing to do here is fire off a bunch of requests and observe some different behaviors. Probably the highest priority are the **EDS requests**, so please try these routes:
http://localhost:5000/eds/degree_days/thawing_index/65.0628/-146.1627
http://localhost:5000/eds/degree_days/freezing_index/63.08/-149.62
http://localhost:5000/eds/degree_days/heating/66.8/-144.1
http://localhost:5000/eds/degree_days/below_zero/62/-146

These responses should mimic those on the currently deployed API - just with different values and different model/scenario names in the `preview` string.

Next, try some **unabridged queries** (no `mmm`-ing):
http://localhost:5000/degree_days/thawing_index/65.0628/-146.1627
http://localhost:5000/degree_days/freezing_index/63.08/-149.62
http://localhost:5000/degree_days/heating/66.8/-144.1
http://localhost:5000/degree_days/below_zero/62/-146

For these routes, observe that the response includes the full time series (1950-2009), except for `daymet`>`historical` which is 1980-2017.

Also, please append `format=csv` to some of those routes and verify that the CSV metadata and filenames and structure are what you'd expect.

Now, try some **default `mmm` summary queries**:
http://localhost:5000/degree_days/thawing_index/65.0628/-146.1627?summarize=mmm
http://localhost:5000/degree_days/freezing_index/63.08/-149.62?summarize=mmm
http://localhost:5000/degree_days/heating/66.8/-144.1?summarize=mmm
http://localhost:5000/degree_days/below_zero/62/-146?summarize=mmm

Note that these queries currently summarize the entire range of historical (1980-2017) and "projected" (1950-2099) data. So, because the historical model runs pre-date the actual historical baseline, you may not see some of trends you might expect. For example, the projected minimum thawing index may be greater than the historical because 1951 was really cold or something. Consider whether or not we'd like the default projected `mmm` summaries to begin at a more recent year, say 2000 or 2018. Also, please try appending `format=csv` to some of those routes and verify that a `mmm` CSV is returned, like for this route:
http://localhost:5000/degree_days/below_zero/62/-146?summarize=mmm&format=csv

OK, next step is to try some **/start_year/stop_year** action - try this route:
http://localhost:5000/degree_days/freezing_index/68/-145/2050/2060

Notice that there is no daymet/historical data returned.
Notice that the year bounds are inclusive, so 11 years worth of data is returned per model-scenario combination.

And verify that the CSV only contains the expected year range as well:
http://localhost:5000/degree_days/below_zero/63/-144/2050/2060?format=csv

For these routes:
http://localhost:5000/degree_days/heating/68/-145/1979/1981
http://localhost:5000/degree_days/heating/68/-145/2016/2018
Notice that you still get model data (because of the historical model runs), but also notice that the historical data just have two years:

For this route:
http://localhost:5000/degree_days/heating/68/-145/1980/2017
Notice that you get the full daymet time series, and that all model-scenarios have data for this time period as well.

Finally, go ahead and **combine start/stop year with `mmm`**:
http://localhost:5000/degree_days/heating/68/-145/2070/2099?summarize=mmm
Notice there is no historical data!

Alternatively, for routes like:
http://localhost:5000/degree_days/heating/68/-145/2000/2020?summarize=mmm
and
http://localhost:5000/degree_days/heating/68/-145/2000/2018?summarize=mmm
notice that there is a historical summary, and that it is identical between the routes because we are only summarizing the available chunk of historical data. And you chain a CSV request arg to these routes too:
http://localhost:5000/degree_days/heating/68/-145/2070/2099?summarize=mmm&format=csv

Whew alright...let me know if you questions or recommendations!

### CODE REVIEW

**Review by cstephen — CHANGES_REQUESTED** [review]
Wow, there is a lot of work represented here, and everything looks nearly perfect! I tested everything in the PR description and also:

- Experimented with different URL year ranges to make sure things behaved as expected.
- Spot-checked that I could find some min/max values from the summary output in the more granular yearly output to make sure it was grabbing the right data for the summaries.
- Verified that a local instance of Arctic-EDS was able to fetch all the expected degree days data from this API branch using the same JSON output structure as before.
- Verified that my degree-days-based ARDAC charts and maps can be updated to work with the new API output and Rasdaman coverages without much hassle.

The updated documentation looks perfect!

Also, while it is a little unexpected, I think it's probably the right move to serve/summarize projected output all the way back to 1950 by default, especially since we provide the ability to request a start/end year manually if desired.

I did find a couple of curiosities, however:

- [ ] <del>The first example URL in the API degree days documentation page (http://localhost:5000/degree_days/heating/65.0628/-146.1627) returns no data, even though it uses the same coordinates as the other degree days endpoints, which do return data. Even stranger, if you add a year range to the URL, it _does_ return data: http://localhost:5000/degree_days/heating/65.0628/-146.1627/1990/2080</del> (This issue fixed itself. May have been a browser cache issue.)
- [ ] I discovered that some years are missing from the JSON output. For example, if you go here: http://localhost:5000/degree_days/below_zero/64.8378/-147.716 and look at the GFDL-ESM2M/rcp85 data, the years 2085 and 2086 are missing. There may be other years missing, and maybe for additional model/scenario combinations also. This is just the first set of missing years I caught.

**Comment by charparr** [comment]
Thanks Craig, appreciate the review! I'll look at these routes that are in question. The first curiosity I need to dig into, know idea what is up there, that does seem odd.

As to the second: my first thought is that there were actually no degree days below zero F for this particular location and time (2085 and 2086) for the GFDL-ESM2M/rcp85 climate future. So, I might need to go back and revisit how to handle this particular flavor of "NoData" and likely revert these values to `0`.

**Comment by charparr** [comment]
OK! Oddity number two was real, and is fixed upstream by https://github.com/ua-snap/ardac-curation/pull/10

I've re-ran that pipeline with modifications, and I'll being rebuilding the rasdaman coverages so we can verify the fix is visible here in the API.

File: postprocessing.py [hunk 1]
```diff
@@ -130,6 +130,24 @@ def prune_nodata_list(data):
     return pruned
 
 
+def prune_nulls_with_max_intensity(data):
```
[the remark this answers is not in the document]
Comment (charparr) [reply postprocessing.py:133]: ✂️ ♾️

**Review by cstephen — APPROVED** [review]
Everything looks good! I've confirmed that the missing (false nodata) years are present in the JSON output, and I've pointed local instances of my ARDAC degree days charts at the new API and everything is working perfectly now. Thanks!!

**Pull request merged**